File: src/core/languages/java/template.java
import java.util.*;
import java.io.*;

// Definition for singly-linked list
class ListNode {
    int val;
    ListNode next;
    ListNode() {}
    ListNode(int val) { this.val = val; }
    ListNode(int val, ListNode next) { this.val = val; this.next = next; }
}

// Definition for binary tree
class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;
    TreeNode() {}
    TreeNode(int val) { this.val = val; }
    TreeNode(int val, TreeNode left, TreeNode right) {
        this.val = val;
        this.left = left;
        this.right = right;
    }
}

class IO {
    private static Scanner scanner = new Scanner(System.in);

    public static class Input {
        // Basic types
        public static boolean readBoolean() { return scanner.nextBoolean(); }
        public static char readChar() { return scanner.next().charAt(0); }
        public static int readInt() { return scanner.nextInt(); }
        public static long readLong() { return scanner.nextLong(); }
        public static float readFloat() { return scanner.nextFloat(); }
        public static double readDouble() { return scanner.nextDouble(); }
        public static String readString() { return scanner.nextLine(); }
        public static String readWord() { return scanner.next(); }
        
        // Utility method to consume newline after primitive inputs
        public static void consumeNewline() { scanner.nextLine(); }
        
        // Arrays
        public static int[] readIntArray() {
            int n = readInt();
            consumeNewline();
            int[] arr = new int[n];
            for (int i = 0; i < n; i++) {
                arr[i] = readInt();
            }
            consumeNewline();
            return arr;
        }
        
        public static long[] readLongArray() {
            int n = readInt();
            consumeNewline();
            long[] arr = new long[n];
            for (int i = 0; i < n; i++) {
                arr[i] = readLong();
            }
            consumeNewline();
            return arr;
        }
        
        public static double[] readDoubleArray() {
            int n = readInt();
            consumeNewline();
            double[] arr = new double[n];
            for (int i = 0; i < n; i++) {
                arr[i] = readDouble();
            }
            consumeNewline();
            return arr;
        }
        
        public static String[] readStringArray() {
            int n = readInt();
            consumeNewline();
            String[] arr = new String[n];
            for (int i = 0; i < n; i++) {
                arr[i] = readString();
            }
            return arr;
        }
        
        public static char[] readCharArray() {
            int n = readInt();
            consumeNewline();
            char[] arr = new char[n];
            for (int i = 0; i < n; i++) {
                arr[i] = readChar();
            }
            consumeNewline();
            return arr;
        }
        
        // 2D Arrays
        public static int[][] readInt2DArray() {
            int m = readInt();
            consumeNewline();
            int[][] arr = new int[m][];
            for (int i = 0; i < m; i++) {
                arr[i] = readIntArray();
            }
            return arr;
        }
        
        public static char[][] readChar2DArray() {
            int m = readInt();
            consumeNewline();
            char[][] arr = new char[m][];
            for (int i = 0; i < m; i++) {
                arr[i] = readCharArray();
            }
            return arr;
        }

        // Collections
        public static List<Integer> readIntList() {
            int n = readInt();
            consumeNewline();
            List<Integer> list = new ArrayList<>();
            for (int i = 0; i < n; i++) {
                list.add(readInt());
            }
            consumeNewline();
            return list;
        }
        
        public static List<String> readStringList() {
            int n = readInt();
            List<String> list = new ArrayList<>();
            consumeNewline();
            for (int i = 0; i < n; i++) {
                list.add(readString());
            }
            return list;
        }
        
        // 2D Lists 
        public static List<List<Integer>> readIntListList() {
            int m = readInt(); 
            IO.Input.consumeNewline();
            List<List<Integer>> result = new ArrayList<>();
            for (int i = 0; i < m; i++) {
                result.add(readIntList());
            }
            return result;
        }
        
        public static List<List<String>> readStringListList() {
            int m = readInt(); 
            IO.Input.consumeNewline();
            List<List<String>> result = new ArrayList<>();
            for (int i = 0; i < m; i++) {
                result.add(readStringList());
            }
            return result;
        }
        
        // Linked List
        public static ListNode readListNode() {
            int n = readInt();
            consumeNewline();
            if (n == 0) return null;
            
            ListNode head = new ListNode(readInt());
            ListNode current = head;
            for (int i = 1; i < n; i++) {
                current.next = new ListNode(readInt());
                current = current.next;
            }
            return head;
        }
        
        // Binary Tree (level-order traversal format)
        public static TreeNode readTreeNode() {
            int n = readInt();
            consumeNewline();
            if (n == 0) return null;
            
            String[] values = new String[n];
            for (int i = 0; i < n; i++) {
                values[i] = readWord();
            }
            
            if (values[0].equals("null")) return null;
            
            TreeNode root = new TreeNode(Integer.parseInt(values[0]));
            Queue<TreeNode> queue = new LinkedList<>();
            queue.offer(root);
            
            int i = 1;
            while (!queue.isEmpty() && i < n) {
                TreeNode node = queue.poll();
                if (i < n) {
                    if (!values[i].equals("null")) {
                        node.left = new TreeNode(Integer.parseInt(values[i]));
                        queue.offer(node.left);
                    }
                    i++;
                }
                if (i < n) {
                    if (!values[i].equals("null")) {
                        node.right = new TreeNode(Integer.parseInt(values[i]));
                        queue.offer(node.right);
                    }
                    i++;
                }
            }
            return root;
        }
    }
    
    public static class Output {
        // Basic types
        public static void write(boolean x) { System.out.print(x ? "true" : "false"); }
        public static void write(char x) { System.out.print("\"" + x + "\""); }
        public static void write(int x) { System.out.print(x); }
        public static void write(long x) { System.out.print(x); }
        public static void write(float x) { System.out.print(x); }
        public static void write(double x) { System.out.print(x); }
        public static void write(String x) { System.out.print("\"" + x + "\""); }
        
        // Arrays
        public static void write(int[] arr) {
            System.out.print("[");
            for (int i = 0; i < arr.length; i++) {
                output(arr[i]);
                if (i < arr.length - 1) System.out.print(",");
            }
            System.out.print("]");
        }
        
        public static void write(long[] arr) {
            System.out.print("[");
            for (int i = 0; i < arr.length; i++) {
                output(arr[i]);
                if (i < arr.length - 1) System.out.print(",");
            }
            System.out.print("]");
        }
        
        public static void write(double[] arr) {
            System.out.print("[");
            for (int i = 0; i < arr.length; i++) {
                output(arr[i]);
                if (i < arr.length - 1) System.out.print(",");
            }
            System.out.print("]");
        }
        
        public static void write(String[] arr) {
            System.out.print("[");
            for (int i = 0; i < arr.length; i++) {
                output(arr[i]);
                if (i < arr.length - 1) System.out.print(",");
            }
            System.out.print("]");
        }
        
        public static void write(char[] arr) {
            System.out.print("[");
            for (int i = 0; i < arr.length; i++) {
                output(arr[i]);
                if (i < arr.length - 1) System.out.print(",");
            }
            System.out.print("]");
        }
        
        // 2D Arrays
        public static void write(int[][] arr) {
            System.out.print("[");
            for (int i = 0; i < arr.length; i++) {
                output(arr[i]);
                if (i < arr.length - 1) System.out.print(",");
            }
            System.out.print("]");
        }
        
        public static void write(char[][] arr) {
            System.out.print("[");
            for (int i = 0; i < arr.length; i++) {
                output(arr[i]);
                if (i < arr.length - 1) System.out.print(",");
            }
            System.out.print("]");
        }
        
        // Collections
        public static void write(List<?> list) {
            System.out.print("[");
            for (int i = 0; i < list.size(); i++) {
                output(list.get(i));
                if (i < list.size() - 1) System.out.print(",");
            }
            System.out.print("]");
        }
        
        // Linked List
        public static void write(ListNode head) {
            System.out.print("[");
            boolean first = true;
            while (head != null) {
                if (!first) System.out.print(",");
                first = false;
                output(head.val);
                head = head.next;
            }
            System.out.print("]");
        }
        
        // Binary Tree (level-order output)
        public static void write(TreeNode root) {
            if (root == null) {
                System.out.print("[]");
                return;
            }
            
            System.out.print("[");
            Queue<TreeNode> queue = new LinkedList<>();
            queue.offer(root);
            boolean first = true;
            
            while (!queue.isEmpty()) {
                int levelSize = queue.size();
                boolean hasNextLevel = false;
                
                for (int i = 0; i < levelSize; i++) {
                    TreeNode node = queue.poll();
                    
                    if (!first) System.out.print(",");
                    first = false;
                    
                    if (node != null) {
                        output(node.val);
                        queue.offer(node.left);
                        queue.offer(node.right);
                        if (node.left != null || node.right != null) {
                            hasNextLevel = true;
                        }
                    } else {
                        System.out.print("null");
                    }
                }
                
                if (!hasNextLevel) break;
            }
            System.out.print("]");
        }
    }
    
    // Convenience wrapper functions
    public static void output(Object x) {
        if (x instanceof Boolean) Output.write((Boolean) x);
        else if (x instanceof Character) Output.write((Character) x);
        else if (x instanceof Integer) Output.write((Integer) x);
        else if (x instanceof Long) Output.write((Long) x);
        else if (x instanceof Float) Output.write((Float) x);
        else if (x instanceof Double) Output.write((Double) x);
        else if (x instanceof String) Output.write((String) x);
        else if (x instanceof int[]) Output.write((int[]) x);
        else if (x instanceof long[]) Output.write((long[]) x);
        else if (x instanceof double[]) Output.write((double[]) x);
        else if (x instanceof String[]) Output.write((String[]) x);
        else if (x instanceof char[]) Output.write((char[]) x);
        else if (x instanceof int[][]) Output.write((int[][]) x);
        else if (x instanceof char[][]) Output.write((char[][]) x);
        else if (x instanceof List) Output.write((List<?>) x);
        else if (x instanceof ListNode) Output.write((ListNode) x);
        else if (x instanceof TreeNode) Output.write((TreeNode) x);
        else System.out.print(x);
    }
}

{{user template}}

{{Solution Class}}

class Main {
    public static void main(String[] args) {
        int t = IO.Input.readInt();
        IO.Input.consumeNewline();
        while (t-- > 0) {
{{Input Statements}}
        }
    }
}

File: src/core/languages/cpp/template.cpp
#include<bits/stdc++.h>
using namespace std;

struct ListNode {
    int val;
    ListNode *next;
    ListNode() : val(0), next(nullptr) {}
    ListNode(int x) : val(x), next(nullptr) {}
    ListNode(int x, ListNode *next) : val(x), next(next) {}
};

struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode() : val(0), left(nullptr), right(nullptr) {}
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
    TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
};

namespace IO {
    namespace Input {
        // Basic types
        void read(bool& x) { cin >> x; }
        void read(char& x) { cin >> x; }
        void read(int& x) { cin >> x; }
        void read(unsigned int& x) { cin >> x; }
        void read(long int& x) { cin >> x; }
        void read(unsigned long& x) { cin >> x; }
        void read(long long int& x) { cin >> x; }
        void read(unsigned long long& x) { cin >> x; }
        void read(float& x) { cin >> x; }
        void read(double& x) { cin >> x; }
        void read(long double& x) { cin >> x; }
        
        // String with whitespace handling
        void read(string& x) { 
            cin >> ws; 
            getline(cin, x); 
        }
        
        // Linked list
        void read(ListNode*& head) {
            int n; 
            cin >> n;
            if (n == 0) { 
                head = nullptr; 
                return; 
            }
            vector<int> values(n);
            for (int i = 0; i < n; i++) { 
                cin >> values[i]; 
            }
            head = new ListNode(values[0]);
            ListNode* current = head;
            for (int i = 1; i < n; i++) {
                current->next = new ListNode(values[i]);
                current = current->next;
            }
        }
        
        // Binary tree (level-order traversal format)
        void read(TreeNode*& root) {
            int n; 
            cin >> n;
            if (n == 0) { 
                root = nullptr; 
                return;
            }
            vector<string> values(n);
            for (int i = 0; i < n; i++) {
                cin >> values[i]; 
            }
            if (values[0] == "null") {
                root = nullptr; 
                return;
            }
            root = new TreeNode(stoi(values[0]));
            queue<TreeNode*> q;
            q.push(root);
            int i = 1;
            while (!q.empty() && i < n) {
                TreeNode* node = q.front();
                q.pop();
                if (i < n) {
                    if (values[i] != "null") {
                        node->left = new TreeNode(stoi(values[i]));
                        q.push(node->left);
                    }
                    i++;
                }
                if (i < n) {
                    if (values[i] != "null") {
                        node->right = new TreeNode(stoi(values[i]));
                        q.push(node->right);
                    }
                    i++;
                }
            }
        }

        template <typename T> void read(vector<T>& x);
        
        // Vector
        template <typename T>
        void read(vector<T>& x) {
            int n;
            cin >> n;
            x.resize(n);
            for (int i = 0; i < n; i++) {
                read(x[i]);
            }
        }
    } // namespace Input
    
    namespace Output {
        // Basic types
        void write(bool x) { cout << (x ? "True" : "False"); }
        void write(char x) { cout << "\"" << x << "\""; }
        void write(int x) { cout << x; }
        void write(unsigned int x) { cout << x; }
        void write(long int x) { cout << x; }
        void write(unsigned long x) { cout << x; }
        void write(long long int x) { cout << x; }
        void write(unsigned long long x) { cout << x; }
        void write(float x) { cout << x; }
        void write(double x) { cout << x; }
        void write(long double x) { cout << x; }
        void write(string x) { cout << "\"" << x << "\""; }
        void write(const char* x) { cout << "\"" << x << "\""; }
        
        // Linked list
        void write(ListNode* head) {
            cout << "[";
            bool first = true;
            while (head) {
                if (!first) cout << ",";
                first = false;
                cout << head->val;
                head = head->next;
            }
            cout << "]";
        }
        
        // Binary tree (level-order output)
        void write(TreeNode* root) {
            if (!root) {
                cout << "[]";
                return;
            }
            cout << "[";
            queue<TreeNode*> q;
            q.push(root);
            bool first = true;
            while (!q.empty()) {
                int levelSize = q.size();
                bool hasNextLevel = false;
                for (int i = 0; i < levelSize; i++) {
                    TreeNode* node = q.front();
                    q.pop();
                    if (!first) cout << ",";
                    first = false;
                    if (node) {
                        cout << node->val;
                        q.push(node->left);
                        q.push(node->right);
                        if (node->left || node->right) hasNextLevel = true;
                    } else {
                        cout << "null";
                    }
                }
                if (!hasNextLevel) break;
            }
            cout << "]";
        }
        
        template <typename T> void write(const vector<T>& x);
        
        // Vector
        template <typename T>
        void write(const vector<T>& x) {
            cout << "[";
            int n = x.size();
            for (int i = 0; i < n; i++) {
                write(x[i]);
                if (i < n - 1) cout << ",";
            }
            cout << "]";
        }
    } // namespace Output
    
    template <typename T>
    void input(T& x) {
        Input::read(x);
    }
    
    template <typename T>
    void output(T x) {
        Output::write(x);
    }
} 

using namespace IO;

{{user template}}

{{Solution Class}}

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    
    int t;
    input(t);
    while (t--) {
{{Input Statements}}
    }
    
    return 0;
}